internal/provider/resource_ably_rule_kinesis.go

```diff
@@ -0,0 +1,433 @@
+
+	// Sets state for the new Ably App.
+	diags = resp.State.Set(ctx, response_values)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read resource
+func (r resourceRule) Read(ctx context.Context, req tfsdk_resource.ReadRequest, resp *tfsdk_resource.ReadResponse) {
+	// Gets the current state. If it is unable to, the provider responds with an error.
+	var state AblyRuleKinesis
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Gets the Ably App ID and Ably Rule ID value for the resource
+	app_id := state.AppID.Value
+	rule_id := state.ID.Value
+
+	// Get Rule data
+	rule, _ := r.p.client.Rule(app_id, rule_id)
+
+	response_values := get_response_values(&rule, state)
+
+	// Sets state to app values.
+	diags = resp.State.Set(ctx, &response_values)
+
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+}
+
+// Update resource
+func (r resourceRule) Update(ctx context.Context, req tfsdk_resource.UpdateRequest, resp *tfsdk_resource.UpdateResponse) {
+	// Gets plan values
+	var plan AblyRuleKinesis
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var state AblyRuleKinesis
+	diags = req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	rule_values := get_plan_values(plan)
+
+	// Gets the Ably App ID and Ably Rule ID value for the resource
+	app_id := state.AppID.Value
+	rule_id := state.ID.Value
+
+	// Update Ably Rule
+	rule, _ := r.p.client.UpdateRule(app_id, rule_id, &rule_values)
+
+	response_values := get_response_values(&rule, plan)
+
+	// Sets state to app values.
+	diags = resp.State.Set(ctx, &response_values)
+
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Delete resource
+func (r resourceRule) Delete(ctx context.Context, req tfsdk_resource.DeleteRequest, resp *tfsdk_resource.DeleteResponse) {
+	// Gets the current state. If it is unable to, the provider responds with an error.
+	var state AblyRuleKinesis
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Gets the Ably App ID and Ably Rule ID value for the resource
+	app_id := state.AppID.Value
+	rule_id := state.ID.Value
+
+	err := r.p.client.DeleteRule(app_id, rule_id)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error deleting Resource",
+			"Could not delete resource, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	// Remove resource from state
+	resp.State.RemoveResource(ctx)
+}
+
+// Import resource
+func (r resourceRule) ImportState(ctx context.Context, req tfsdk_resource.ImportStateRequest, resp *tfsdk_resource.ImportStateResponse) {
+	tfsdk_resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
```

Comment:
Should this be appid+id like the other resources?

Reply:
Modified